### PR TITLE
Rails 5 support patches

### PR DESCRIPTION
**Pull request by samphippen**

### CODE REVIEW

**Comment by samphippen** [comment]
This PR is green against specs, but not against our acceptance tests.

File: spec/rspec/rails/matchers/have_http_status_spec.rb [hunk 1]
```diff
@@ -382,12 +382,12 @@ def create_response(opts = {})
...
       subject(:matcher) { have_redirect_status }
 
-      let(:code) { 333 }
+      let(:code) { 308 }
     end
 
     describe "matching a response" do
       it "returns true for a response with a 3xx status code" do
-        any_3xx_code = 333
+        any_3xx_code = 308
```
Comment (JonRowe) [inline spec/rspec/rails/matchers/have_http_status_spec.rb]: Shouldn't these be dynamic depending on version? Or does 308 play ok for rails 4 etc too?

Comment (samphippen) [inline spec/rspec/rails/matchers/have_http_status_spec.rb]: 308 is actually a valid code in the spec. 333 is not.

Comment (JonRowe) [inline spec/rspec/rails/matchers/have_http_status_spec.rb]: Ah so unrelated change?

Comment (samphippen) [inline spec/rspec/rails/matchers/have_http_status_spec.rb]: The reason I made this change is that somewhere along the way, rails 5 no-longer accepts 3xx codes that aren't part of the spec in a bunch of places. 333 is not part of the spec. Given that all old versions of rails support any http response code that is in the spec, this seemed like the easiest change to make for this PR to work.

Comment (JonRowe) [inline spec/rspec/rails/matchers/have_http_status_spec.rb]: :+1:

**Comment by samphippen** [comment]
After a pairing session with @sgrif we're making some progress on this. We'll be spending a few more evenings on it to make progress. Things that definitely work at the moment:

* Installing RSpec on rails master
* The rails controller testing gem
* our spec suite
* Our smoke suite (with and without AR)

Things that don't work:

* Our cuke suite

File: spec/rspec/rails/matchers/have_http_status_spec.rb (hunk 1 above)
Comment (samphippen) [inline spec/rspec/rails/matchers/have_http_status_spec.rb]: **This has been resolved**

**Comment by sgrif** [comment]
Other things that don't work: My dotfiles. :trollface:

**Comment by samphippen** [comment]
@sgrif and myself just saw this go green on rails master. Going to do my best to get it all-supported-versions compatible soon.

**Comment by samphippen** [comment]
This is fully backwards compatible, rails 3, 4 and 5 support. Do not merge this yet. I have some rebasing/other work to do.

**Comment by benoittgt** [comment]
Thanks @samphippen and @sgrif

**Comment by samphippen** [comment]
@JonRowe @cupakromer I've just fixed up the history here. Would you mind taking a look through the commits and pointing out any changes that aren't made clear by the commit messages?

**Comment by sgrif** [comment]
Since beta1 is out, should this point at the released gems rather than git?

**Comment by samphippen** [comment]
@sgrif really, it should be beta1 and master.

**Comment by sgrif** [comment]
@samphippen My thinking here is while nothing should break before release, if it is we don't want your build failing do we?

**Comment by samphippen** [comment]
@sgrif yeah, I'm going to keep master in the allowed failures section, and add beta1 right now.

**Comment by sgrif** [comment]
:+1:

File: .travis.yml [hunk 2]
```diff
@@ -56,6 +56,8 @@ matrix:
     # Rails 5.x only supports 2.2
```
Comment (l4u) [inline .travis.yml]: Should this be 2.2.2+ instead of 2.2?

**Comment by JonRowe** [comment]
:+1:

**Comment by samphippen** [comment]
@JonRowe I intentionally wasn't merging this unless we needed any further patches before rails went stable. Is it OK if I revert the merge and restore the branch?

**Comment by JonRowe** [comment]
I guess but I'd rather people got this sooner rather than later, seems strange to have to get people on this branch and then keep rebasing simply because rails isn't released.

**Comment by samphippen** [comment]
it makes it easy to track all the changes for rails in one place, which is my concern. There also wasn't a changelog entry yet...

**Comment by samphippen** [comment]
I think on balance, having this merged is fine, actually.

**Comment by JonRowe** [comment]
It's not the only place though (as I had initial work elsewhere) and makes it harder for others to contribute :) Happy for you to add a changelog direct to master

**Comment by JuanitoFatas** [comment]
If anyone trying rspec-rails with Rails 5 like me, you need these: 

```ruby
gem "rspec-rails", git: "https://github.com/rspec/rspec-rails.git", branch: "master"
gem "rspec-core", git: "https://github.com/rspec/rspec-core.git", branch: "master"
gem "rspec-support", git: "https://github.com/rspec/rspec-support.git", branch: "master"
gem "rspec-expectations", git: "https://github.com/rspec/rspec-expectations.git", branch: "master"
gem "rspec-mocks", git: "https://github.com/rspec/rspec-mocks.git", branch: "master"
```

**Comment by AbdullahAlger** [comment]
Thanks @JuanitoFatas! I was getting all kinds of errors before this.

**Comment by CodedBeardedSignedTaylor** [comment]
Thanks for the tip, @JuanitoFatas!
